ocaml/xenopsd/xc/mem_stats.ml

```diff
@@ -0,0 +1,391 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = "mem_stats" end)
+
+module Compat = struct
+  (* backporting support *)
+
+  (*
+  open Stdext
+  let mtime_clock_counter = Mtime.counter
+  let mtime_clock_count = Mtime.count
+  let mtime_span_to_s = Mtime.to_s
+  let file_lines_fold = Stdext.Unixext.file_lines_fold
+  let mutex_execute = Stdext.Threadext.Mutex.execute
+
+  open Rrdd_plugin
+  let reporter_cache : Reporter.t option ref = ref None
+  let reporter_async ~shared_page_count ~dss_f =
+    Reporter.start_async
+      (module D : Debug.DEBUG)
+      ~uid:"mem-stats"
+      ~neg_shift:0.5
+      ~target:(Reporter.Local shared_page_count)
+      ~protocol:Rrd_interface.V2
+      ~dss_f
+  let reporter_cancel = Reporter.cancel
+  *)
+
+  let mtime_clock_counter = Mtime_clock.counter
+
+  let mtime_clock_count = Mtime_clock.count
+
+  let mtime_span_to_s f = Mtime.Span.to_float_ns f *. 1e-9
+
+  let file_lines_fold = Xapi_stdext_unix.Unixext.file_lines_fold
+
+  let mutex_execute = Xapi_stdext_threads.Threadext.Mutex.execute
+
+  let reporter_async ~shared_page_count ~dss_f =
```

Comment:
there is already a `start_async` in rrdd_plugin.ml, why not use that one?

Reply:
I think I didn't notice it, updated.